src/shim.js

```diff
@@ -170,18 +185,8 @@ function executeCommand(command) {
     } else if (objectSpace[command.target]) {
         const target = objectSpace[command.target];
         const method = target[command.name];
-
-        if (haveCallbacks.indexOf(command.name) === -1) {
-            command.response = method.apply(target, command.params);
-            completeCommand(command);
-        } else {
-            let params = command.params.slice(); // copy params
-            params.push(status => {
-                command.response = status;
-                completeCommand(command);
-            });
-            method.apply(target, params);
-        }
+        command.response = method.apply(target, command.params);
+        completeCommand(command);
```

Comment:
Thanks! [The `else if` statement that follows](https://github.com/amir20/phantomjs-node/blob/master/src/shim.js#L185-L190) is in 'theory' deprecated, but the 'Page #reject(...) works when there is an error' test fails if the block is removed.